test/java/com/arpnetworking/metrics/portal/integration/controllers/KairosDbProxyControllerIT.java

```diff
@@ -167,8 +168,74 @@ public void testTagQuery() throws ExecutionException, InterruptedException {
         assertTrue(result.getValues().isEmpty());
     }
 
+    @Test
+    public void testQueryMetricsExcludeFilterTagInResponse() throws ExecutionException, InterruptedException {
+        _kairosDbClientDirect.addDataPoints(ImmutableList.of(
+                new MetricDataPoints.Builder()
+                        .setName(_metricName)
+                        // NOTE: KDB requires every metric to have at least one tag!
+                        .setTags(ImmutableMap.of(
+                                "host", "host.example.com",
+                                "ignored", "foo"))
```

Comment:
I'm not sure I understand this comment (or the purpose of this test) -- if you didn't set the `host` tag on the datapoint, would we throw because all the remaining tags are excluded?

Reply:
So the comment is a warning that all metrics added to KDB must have at least one tag set or the REST API of KDB will reject them. That was surprising to me, so I started calling it out in places. In this particular test, I am adding two tags so it's not really a concern.

Further, tags are _never_ filtered on  data add. In fact, all the data add calls use the `_kairosDbClientDirect` which bypasses our logic in MP and hits KDB directly. In fact, MP is a read-only proxy for KDB and cannot be used to write data to KDB.

The purpose of these two tests is that if data is written to KDB with a tag that we want to ignore, that tag is _NOT_ returned implicitly when a query for that metric is issued (KDB returns tags for metrics if there is just one value for the tag in the query result set). However, that tag SHOULD be returned if the query explicitly sets the tag on the filter even if it is configured to be ignored.

Reply:
Ah, I see, because `assertMetricsQuery` asserts that there's only one tag. Got it.